pandas/core/arrays/datetimes.py

```diff
@@ -1085,3 +1203,91 @@ def to_julian_date(self):
 
 
 DatetimeArrayMixin._add_comparison_ops()
+DatetimeArrayMixin._add_datetimelike_methods()
+
+
+def _generate_regular_range(cls, start, end, periods, freq):
+    if isinstance(freq, Tick):
+        stride = freq.nanos
+        if periods is None:
+            b = Timestamp(start).value
+            # cannot just use e = Timestamp(end) + 1 because arange breaks when
+            # stride is too large, see GH10887
+            e = (b + (Timestamp(end).value - b) // stride * stride +
+                 stride // 2 + 1)
+            # end.tz == start.tz by this point due to _generate implementation
+            tz = start.tz
+        elif start is not None:
+            b = Timestamp(start).value
+            e = b + np.int64(periods) * stride
+            tz = start.tz
+        elif end is not None:
+            e = Timestamp(end).value + stride
+            b = e - np.int64(periods) * stride
+            tz = end.tz
+        else:
+            raise ValueError("at least 'start' or 'end' should be specified "
+                             "if a 'period' is given.")
+
+        data = np.arange(b, e, stride, dtype=np.int64)
+        data = cls._simple_new(data.view(_NS_DTYPE), None, tz=tz)
+    else:
+        tz = None
+        if isinstance(start, Timestamp):
+            tz = start.tz
+            start = start.to_pydatetime()
+
+        if isinstance(end, Timestamp):
+            tz = end.tz
+            end = end.to_pydatetime()
+
+        xdr = generate_range(start=start, end=end,
+                             periods=periods, offset=freq)
+
+        dates = list(xdr)
+        # utc = len(dates) > 0 and dates[0].tzinfo is not None
+        values = np.array([x.value for x in dates])
+        data = cls._simple_new(values, freq=freq, tz=tz)
```

Comment:
This is the second change from the existing version.  ATM this reads:

```
        # utc = len(dates) > 0 and dates[0].tzinfo is not None
        data = tools.to_datetime(dates)
```

Reply:
Maybe you can go ahead and blow away the commented line of code above and just directly use `xdr`:

`values = np.array([x.value for x in xdr])`

Reply:
Good idea, will do.  I'd like to track down the source of the comment and make sure it isn't something important.

Reply:
hmm this list conversion is problematic